sdk/src/Core/Amazon.Runtime/AmazonServiceClient.cs

```diff
@@ -344,13 +270,11 @@ protected static TResponse EndInvoke<TResponse>(IAsyncResult result)
 
         protected void ProcessPreRequestHandlers(IExecutionContext executionContext)
         {
-            //if (request == null)
-            //    return;
-            if (mBeforeMarshallingEvent == null)
+            if (BeforeMarshallingEvent == null)
                 return;
 
             PreRequestEventArgs args = PreRequestEventArgs.Create(executionContext.RequestContext.OriginalRequest);
-            mBeforeMarshallingEvent(this, args);
+            BeforeMarshallingEvent(this, args);
```

Comment:
💡 The recommended event invocation pattern is:

```csharp
BeforeMarshallingEvent?.Invoke(this, PreRequestEventArgs.Create(executionContext.RequestContext.OriginalRequest));
```

Note that `?.` triggers an early escape when null; i.e. `PreRequestEventArgs.Create` will only be called for this expression if one or more event handlers is registered.

Reply:
📝 This cannot be directly applied to all cases, but instead shows one example to consider in a future PR.

Reply:
I've updated the one case where we can apply this.